simphony_ui/ui.py

```diff
@@ -33,12 +49,38 @@ class Application(HasStrictTraits):
 
     @on_trait_change('run_button')
     def run_calc(self):
+        if self.calculation_running:
+            raise BaseException('Calculation already running...')
+        self.calculation_running = True
+        self.progress_dialog.open()
+        future = self._executor.submit(self._run_calc_threaded)
+        future.add_done_callback(self._calculation_done)
+
+    def _run_calc_threaded(self):
```

Comment:
docstring. Make it very clear that this function is running in the secondary thread.